dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -602,6 +609,14 @@ private void updateInfo(Workflow oldWorkflow, Workflow newWorkflow) {
 
         List<Token> tokens = checkOnZenodoToken(user);
         Token zenodoToken = Token.extractToken(tokens, TokenType.ZENODO_ORG);
+
+        // Update the zenodo token in case it changed. This handles the case where the token has been changed but an error occurred, so the token in the database was not updated
+        if (zenodoToken != null) {
+            tokenDAO.update(zenodoToken);
+        }
+        sessionFactory.getCurrentSession().getTransaction().commit();
```

Comment:
This feels a little odd to me -- the UnitOfWork annotation starts one transaction but commits/rolls back a different one. I guess the only alternative is to remove the UnitOfWork annotation and programmatically start one TX at the beginning of the method. I lean towards that, but don't feel that strongly.

Also, what happens if the commit fails (extremely unlikely, but it could)? It will throw a RollbackException -- how will UnitOfWork react? I guess worst case it will still be an error that the user sees.

Alternative suggestion:

If you keep it this way, only commit/start the transaction inside the if block

Reply:
undecided, I think you might need UnitOfWork to get a session at all, but you might be able to try @UnitOfWork(transactional=false)